R/track-util.R

```diff
@@ -0,0 +1,142 @@
+#' track helpers
+#' 
+#' These functions allow you to calculate different migration parameters. 
+
+#' * `direction`: calculate the mean direction of movement
+#' * `displacement`: displacement as 2-norm(last_position - first_position)
+#' * `distance_traveled`: calculate the accumlated distance the cell has traveled 
+#' * `life_time`: temporal length of the cell (last frame - first frame +1) 
+#' * `speed()`: calculate the speed in px per Frame
+#' * `valid_tracks`:
+#' * `xfmi`: calculate foward migration index in x direction
+#' * `yfmi`: calculate foward migration index in x direction
+
+#' 
+#' @param 
+#' @name track_helpers
+#' @return 
+#' 
+#' @importFrom magrittr %>%
+#' @importFrom magrittr %<>%
+NULL
+
+# make sure to not disturb the environment ------------------------------------
+old <- options(stringsAsFactors = FALSE)
+on.exit(options(old), add = TRUE)
+old <- setwd(tempdir())
+on.exit(setwd(old), add = TRUE)
+
+#' @export
+#' @rdname select_helpers
+
+
+add_dist <- function(tracks, group_track){
+  dplyr::right_join(
+    tracks %>%
+      dplyr::select_(.dots = c(group_track, 'Location_Center_X', 'Location_Center_Y','Metadata_timePoint')) %>%
+      dplyr::mutate(Metadata_timePoint2 =  (Metadata_timePoint - 1) ) %>% 
+      dplyr::filter(Metadata_timePoint2 != -1) %>%
+      dplyr::select(-Metadata_timePoint), 
+    tracks, by = (.dots = c(group_track, "Metadata_timePoint2" = "Metadata_timePoint" ))) %>%
+    dplyr::mutate(Track_dX = Location_Center_X.x - Location_Center_X.y) %>% 
+    dplyr::mutate(Track_dY = Location_Center_Y.x - Location_Center_Y.y) %>% 
+    dplyr::select(-Location_Center_X.x, -Location_Center_Y.x) %>%
+    dplyr::rename(Location_Center_X = Location_Center_X.y) %>%
+    dplyr::rename(Location_Center_Y = Location_Center_Y.y) %>% 
+    dplyr::rename(Metadata_timePoint = Metadata_timePoint2) %>%
+    dplyr::mutate(TrackObjects_Distance_Traveled = sqrt(Track_dX^2 + Track_dY^2)) 
+}
+
+get_track_speed <- function(tracks){
+  tracks %>%
+    dplyr::summarize(Track_Length = n(),
+      Track_Speed = sum(TrackObjects_Distance_Traveled, na.rm = TRUE) / (n() - 1), 
+      Track_Speed_max = max(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Speed_X = sum(Track_dX, na.rm = TRUE) / (n() - 1),
+      Track_Speed_Y = sum(Track_dY, na.rm = TRUE) / (n() - 1)) %>%
+    dplyr::select(-Track_Length)
+}
+
+get_track_fmi <- function(tracks) {
```

Comment:
You expose a track namespace. You don’t need to prefix your identifiers with “track.”